core/string/ustring.cpp

```diff
@@ -2575,6 +2575,72 @@ int64_t String::to_int(const wchar_t *p_str, int p_len) {
 	return integer * sign;
 }
 
+uint64_t String::to_uint64(bool p_clamp) const {
+	int str_length = length();
+	if (str_length == 0) {
+		return 0;
+	}
+
+	int start = 0;
+	// Skip leading whitespace
+	while (start < str_length &&
+			(is_whitespace(operator[](start)) || operator[](start) == '+')) {
+		start++;
+	}
+
+	// Check for negative sign
+	if (start < str_length && operator[](start) == '-') {
+		if (!p_clamp) {
+			ERR_FAIL_V_MSG(0,
+					"Cannot represent " + *this + " as a 64-bit unsigned integer, since the value is negative.");
+		}
+		return 0;
+	}
+
+	uint64_t integer = 0;
```

Comment:
Your current implementation will parse (post-stripped) empty strings (like "+ " ) as 0. I don't think this is wanted?

Reply:
This was again intentional of parse the number and then drop anything that isn't the number after it like a decimal, space, new line or any other junk.